src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -345,6 +351,33 @@ protected async Task RemoveConnectionAsync(IServiceConnection c, bool migratable
             Log.TimeoutWaitingForFinAck(Logger, retry);
         }
 
+        internal bool GetServiceStatus(bool active, int checkWindow, TimeSpan checkTimeSpan)
+        {
+            if (active)
+            {
+                _inactiveInfo = (0, null);
+                return true;
+            }
+            else
+            {
+                var info = _inactiveInfo;
+                var last = info.last ?? DateTime.UtcNow;
+                var count = info.count;
+                count++;
+                _inactiveInfo = (count, last);
```

Comment:
not thread-safe, could multiple status ping near similar time turns out to be one?

Reply:
maybe, but overwrites does not matter

Reply:
That will affect the result about count and even about active/inactive check, like inactive may be delayed to know. Is this expected and acceptable?